tests/test_cli.py

```diff
@@ -363,18 +363,23 @@ def test_lint_log_user_warning(self, mock_lint, mock_is_pipeline):
     def test_schema_lint(self, mock_get_schema_path):
         """Test nf-core schema lint defaults to nextflow_schema.json"""
         cmd = ["schema", "lint"]
-        result = self.invoke_cli(cmd)
-        assert mock_get_schema_path.called_with("nextflow_schema.json")
-        assert "nextflow_schema.json" in result.output
+        Path("nextflow_schema.json").touch()
```

Comment:
Would it be better to use `with self.runner.isolated_filesystem():` here before we create the temporary file?

This would avoid any issues with unintended consequences or cleanup.

Reply:
hmm, can't find any info about this, can you point to the docs for this (because it would be helpful to have them isolated also for other tests).

Reply:
[Click docs](https://click.palletsprojects.com/en/8.1.x/testing/#file-system-isolation)

Example of working usage:
```python
@mock.patch("nf_core.schema.PipelineSchema.get_schema_path")
def test_schema_lint(self, mock_get_schema_path):
    """Test nf-core schema lint defaults to nextflow_schema.json"""
    cmd = ["schema", "lint"]
    with self.runner.isolated_filesystem():
        with open("nextflow_schema.json", "w") as f:
            f.write("{}")
        result = self.invoke_cli(cmd)
        mock_get_schema_path.assert_called_with("nextflow_schema.json")
        assert "nextflow_schema.json" in result.output
```